docs/guide/tutorials/task-guide.md

```diff
@@ -0,0 +1,524 @@
+
+For example, the following Task design would be bad and could not be deployed to Cloud nor could it run using a Dask-based executor:
+
+```python
+import google.bigquery
+
+class BigQuery(Task):
+    def __init__(self, *args, **kwargs):
+        self.client = bigquery.Client() # <-- not serializable
+        super().__init__(*args, **kwargs)
+
+    def run(self, query: str):
+        self.client.query(query)
+```
+In this case, instantiating the Client outside of the `run` method will prevent this Task from being serializable by `cloudpickle` (instead you should instantiate the client _within_ the `run` method).
+
+#### Avoid Statefulness
+
+In addition to avoiding attributes which cannot be serialized, you should also avoid on relying on statefulness in your Tasks.  For example, in designing a Task class, relying on an attribute which stores state would not behave as expected:
+```python
+class Bad(Task):
+    def __init__(self, *args, **kwargs):
+        self.run_count = 0
+        super().__init__(*args, **kwargs)
+
+    def run(self):
+        self.run_count += 1
+        return self.run_count
+```
+
+Similarly, relying on some global state will ultimately result in headache:
+
+```python
+global_state = {"info": []}
+
+@task
+def task_one():
+    global_state["info"].append(1)
+
+@task
+def task_two():
+    global_state["info"].append(2)
+```
+
+#### Use a Task class for "templating" Tasks
+
+Using the [subclass approach](subclassing-the-task-class) approach to desiging Tasks can be beneficial whenever you want to provide a configurable task "template", whose default values can be both set at initialization time and optionally overwritten at runtime.  For example, let's alter the `add_task` we created above to provide a default value for `y`:
+
+```python
+class AddTask(Task):
+    def __init__(self, *args, y=None, **kwargs):
+        self.y = y
+        super().__init__(*args, **kwargs)
+
+    def run(self, x, y=None):
+        if y is None:
+            y = self.y
+
+        return x + y
+
+
+add_task = AddTask(y=0)
+
+with Flow("add-with-default") as f:
+    result_one = add_task(x=1)
+    result_two = add_task(x=0, y=10)
+```
+
+We've found this pattern of setting defaults which are optionally overwritten at runtime to be so common, we created a [utility function to minimize boilerplate](https://docs.prefect.io/api/unreleased/utilities/tasks.html#prefect-utilities-tasks-defaults-from-attrs).  In addition, subclassing allows you to write custom class methods that are organized in one place.
+
+::: warning Always call the parent Task initialization method
+Anytime you subclass `Task`, _make sure to call the parent initialization method_!  This ensures Prefect will recognize your custom Task as an actual Task.  In addition, we highly recommend always allowing for arbitrary keyword arguments which are passed to the Task `__init__` method.  This ensures that you can still set things such as Task tags, custom names, result handlers, etc.
```

Comment:
_In addition, we highly recommend always allowing for arbitrary keyword arguments_

For added clarity can this just mention that it is `**kwargs`?

Reply:
Def